contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,248 @@
+
+For an example, let&#39;s say we have 3 federated clusters and the following yamls are submitted to create a federated statefulset:
+```
+---
+apiVersion: v1
+kind: Service
+metadata:
+ name: nginx
+ labels:
+   app: nginx
+spec:
+ ports:
+ - port: 80
+   name: web
+ clusterIP: None
+ selector:
+   app: nginx
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+ name: web
+spec:
+ serviceName: "nginx"
+ replicas: 5
+ template:
+   metadata:
+     labels:
+       app: nginx
+   spec:
+     terminationGracePeriodSeconds: 10
+     containers:
+     - name: nginx
+       image: gcr.io/google\_containers/nginx-slim:0.8
+       ports:
+       - containerPort: 80
+         name: web
+       volumeMounts:
+       - name: www
+         mountPath: /usr/share/nginx/html
+ volumeClaimTemplates:
+ - metadata:
+     name: www
+   spec:
+     accessModes: ["ReadWriteOnce"]
+     resources:
+       requests:
+         storage: 1Gi
+```
+The service will be created in each of the 3 clusters unmodified.
+
+The default replica partitioning will mean a distribution of **2, 2 and 1** replicas to the 3 clusters.
+
+The statefulset spec submitted to the clusters will have the statefulset metadata.name modified to _**web+&lt;cluster-name&gt;**_.
+
+Using the example above, with statefulset name _web_, service name _nginx_ and domain name as _federation.example_ the dns names created against each instance would be as below, discoverable both locally and globally (assuming the registered cluster names being c1, c2 and c3):
+```
+web-c1-0.nginx.mynamespace.svc.federation.example
+web-c1-1.nginx.mynamespace.svc.federation.example
+web-c2-0.nginx.mynamespace.svc.federation.example
+web-c2-1.nginx.mynamespace.svc.federation.example
+web-c3-0.nginx.mynamespace.svc.federation.example
+```
+This above is fine for the case of multiple local quora (one in each cluster), but these identities cannot work across the clusters, as the local pod IPs cannot be reached across the clusters.
+
+Similar to _design alternative 1_, the proposal to make it work across the clusters is to create an additional service of type &#39;LoadBalancer&#39; for each pod instance that is created in the clusters. This will be automatically created by the federated statefulset controller, on demand, proposed to be controlled by an annotation on the statefulset.
+If the annotation specifies so (users choice), the federation controller will create the additional service of type &#39;LoadBalancer&#39; against each pod replica.
+The service name for service, one each against the pod will be the same name as the pod identity.
+The LB service creation at federation has the same drawback as discussed in _design alternative 1_.
+The proposed solution is elaborated in section **Handling additional LB Services**.
+
+## Storage volumes
+
+There are two ways a federated statefulset can be assigned a persistent storage.
```

Comment:
I am sorry; I did not get you here.. ? You mean test against storage classes.. ?